pkg/attestation/crafter/runners/oidc/gitlab.go

```diff
@@ -95,6 +100,24 @@ func parseToken(ctx context.Context, providerURL string, tokenString string) (*G
 		return nil, fmt.Errorf("failed to extract claims: %w", err)
 	}
 
+	// Convert raw token to JWT token to use VerifyAudience
+	// Parse the JWT token without validation (we already validated it with the OIDC provider)
+	parsedToken, _, err := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
```

Comment:
do you need to parse it? At this point I am not sure `VerifyAudience` makes sense if it requires all these changes. Did you look at what the method does inside to see if it's worth it?

Reply:
can't you use `idtoken` to get the claims? You seem to be doing smth already?

```
var claims map[string]interface{}
	if err := idToken.Claims(&claims); err != nil {
		return nil, fmt.Errorf("failed to extract claims: %w", err)
	}
```

Reply:
In theory, just `idToken.Audience` would return the list of audience values. So something like this would also work:
```
strings.Contains(idToken.Audience, ExpectedAudience)
```

Reply:
Yes, but @migmartri wanted to move away from the simple checking ;)

Reply:
and note that `strings.Contains(idToken.Audience, ExpectedAudience)` might not be valid since it's checking a substring, so for example 

`hacking-chainloop` will patch the `chainloop` audience?

Reply:
Ok, but not that `verifier.Verify` is already doing the claim extraction for you into `idToken.Audience` field. https://github.com/coreos/go-oidc/blob/v3/oidc/verify.go#L250

Reply:
ok, up to you guys, this is the method I was talking about which we use in all our middlewares today https://pkg.go.dev/github.com/golang-jwt/jwt/v4@v4.5.2#RegisteredClaims.VerifyAudience

Reply:
Let's do it like this - I'll have a look tomorrow if I can simplify the code and adjust and merge after.

Reply:
I've looked through the code in the `verify.go` in `go-oidc` - I think we can compare slices, the additional verification is already done. I'll modify the code and merge.